tests/test_aac_vault.py

```
@@ -0,0 +1,212 @@
+        vault = AacVault(token=None, aac_path=fake_aac)
+        vault.start()
+        assert vault._paired is False
+        vault.stop()
+
+    def test_pairing_failure_raises(self, fake_aac_fail: str) -> None:
+        vault = AacVault(token="ABC-DEF-123", aac_path=fake_aac_fail)
+        with pytest.raises(RuntimeError, match="aac pairing failed"):
+            vault.start()
+
+    def test_reads_token_from_env(self, fake_aac: str) -> None:
+        os.environ["AAC_TOKEN"] = "ENV-TOK-123"
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to: "AAC_TOKEN"

(S105)